cargo-geiger/src/tree/traversal/dependency_node.rs

```diff
@@ -82,14 +82,44 @@ fn construct_dependency_type_nodes_hashmap<'a>(
         };
 
         dependency_type_nodes
-            .get_mut(&edge.weight().to_cargo_core_dep_kind())
+            .get_mut(&PrivateDepKind::from_dependency_kind(&edge.weight()))
             .unwrap()
             .push(dependency.clone());
     }
 
     dependency_type_nodes
 }
 
+// cargo_metadata::DependencyKind doesn't implement Eq or Hash, and so can't
```

Comment:
Make an upstream PR or issue, if you have the time.

Reply:
Sounds good! Have landed https://github.com/oli-obk/cargo_metadata/pull/148 to resolve